code/datums/gamemode/role/time_agent.dm

```diff
@@ -274,33 +359,75 @@
 
 /obj/item/device/timeline_eraser
 	name = "timeline eraser"
-	desc = "Erase someone from the timeline. It has an unusual affinity against time travellers..."
+	desc = "A strange button."
 	icon_state = "jump_charge"
 	w_class = W_CLASS_SMALL
-	var/channeling = FALSE
+	flags = TIMELESS
+	var/in_process = FALSE
+	var/charge = 5 //Will set to 0 and gradually increment to from 0 to 5 (10 seconds total), after which it can be used again
+	var/logo
 
+/obj/item/device/timeline_eraser/examine(mob/user, size, show_name)
+	..()
+	if(istimeagent(user))
+		to_chat(user, "[logo] <span class='notice'>As a time agent, you know that this device can erase nearly anything from reality. Erasing entities will take 10 seconds, erasing objects will take 5 seconds and erasing other time agents will take no time at all. People with temporal suits are protected from its effects.</span>")
+		if(charge < 5)
+			to_chat(user, "<span class='warning'>It is still recharging.</span>")
 
+/obj/item/device/timeline_eraser/New()
+	..()
+	logo = "<img src='data:image/png;base64,[icon2base64(icon('icons/logos.dmi', "time-logo"))]'/>"
+
+/obj/item/device/timeline_eraser/Destroy()
+	processing_objects.Remove(src)
+	..()
 
-/obj/item/device/timeline_eraser/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
-	var/duration = 100
+/obj/item/device/timeline_eraser/process()
+	charge++
+	if(charge >= 5)
+		icon_state = "jump_charge"
+		processing_objects.Remove(src)
+
+/obj/item/device/timeline_eraser/preattack(atom/target, mob/user, proximity_flag)
+	if(!istimeagent(user))
+		return ..()
+	if(!proximity_flag)
+		return
+	if(in_process)
+		to_chat(user, "<span class='warning'>\The [src] is already erasing someone from reality!</span>")
+		return
+	if(charge < 5)
+		to_chat(user, "<span class='warning'>\The [src] is still recharging!</span>")
+		return
+	. = 1
+	var/duration = 100 //10 seconds
 	// TODO: Make the timestop, properly stop when the process is done
 	if(istype(target, /mob))
 		var/mob/M = target
 		if(istimeagent(M))
 			duration = 0
+	if(istype(target, /obj))
+		duration = 50 //5 seconds
 
-	if(proximity_flag && !(target.flags & TIMELESS))
-		channeling = TRUE
-		spawn()
-			while(channeling)
-				//make the process obvious, but don't impede people interfering with it too much
-				timestop(src, 30, 7, 0, /mob/living/carbon/)
-				sleep(60)
-		if(do_after(user, target, duration))
-			delete_from_timeline(target, user)
-		channeling = FALSE
+	if(istype(target, /obj/item/device/jump_charge)) //It is already unaffected but leaving a message here for trying to soft-lock is funny
+		to_chat(user, "<span class='warning'>Are you insane?</span>")
+		return
```

Comment:
shouldn't this be with the exits at the top?
also make it more descriptive because people might click their own jump charge by accident and not know what caused that message

Reply:
Make it play [this] (https://youtu.be/OYa5aQb3YGE?t=28)if you try.

Reply:
I can't put a goofy sound to it yet @Eneocho